src/App.tsx

```diff
@@ -8,7 +8,37 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+import { FilterType, filterTodos } from './utils/filter';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | null>(null);
+  const [activeTodo, setActiveTodo] = useState<Todo | null>(null);
+  const [todoType, setTodoType] = useState(FilterType.ALL);
+  const [query, setQuery] = useState('');
+
+  const fetchData = async () => {
+    const filteredTodos = await filterTodos(todoType, query);
+
+    setTodos(filteredTodos);
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, [query, todoType]);
```

Comment:
You don't need to make API call after every `query` or `todoType` change
You should fetch them 1 time, save them to state. and then just filter